src/CMakeLists.txt

```diff
@@ -169,8 +170,10 @@ endif()
 
 
 install( DIRECTORY ../examples/
-         DESTINATION examples )
+         DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/${CMAKE_PROJECT_NAME}/examples )
 
 install( DIRECTORY ../scripts/
-         DESTINATION scripts )
+         DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/${CMAKE_PROJECT_NAME}/scripts )
 
+install( FILES ${SCHEMA_DIR}/schema.xsd
+         DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/${CMAKE_PROJECT_NAME}/schema )
```

Comment:
The creation of `schema.xsd` is done iff `ENABLE_XML_UPDATES AND ENABLE_MPI AND UNIX AND NOT CMAKE_HOST_APPLE AND NOT ENABLE_CUDA` evaluates to true. Otherwise we're using the last existing version of `schema.xsd` (yes, we commit generated code and I do agree that we shouldn't...).

Can you manage to reunite the creation of `schema.xsd` and its installation. I do not think we want a mismatch between `geosx` and its scheme.

Reply:
The root of the issue is that, yes, we commit the _generated_ file `schema.xsd`, and I claim that it's a bad pattern (I think mostly driven by some `readthedocs` deployment considerations).

As such, there will always be a moment where the `schema.xsd` will be out of sync (Because someone will have forgotten to commit, for example. Or because the dev won't have `ENABLE_XML_UPDATES` set to `true` to save time. Or because some `CLion` remote working dev pattern won't update `schema.xsd` in your work tree, but only on the remote and you may not notice).

To enforce sync, we should only deploy the `schema.xsd` that has been generated by the `geosx` binary itself.
That way, our end-users will always have something reliable.

Reply:
In that case any deployment of GEOSX on a GPU system will not have a schema installed... which is probably not what we want? Generation is only disabled with CUDA to avoid producing diffs on LvArray type names (something developers care about), but users need the schema regardless (whether freshly generated or pre-committed) for input validation. I agree that neither solution is perfect and we should ideally improve the schema generation to not produce diffs between systems/configurations.

Reply:
> In that case any deployment of GEOSX on a GPU system will not have a schema installed... which is probably not what we want?

What I do not want is that end-users get provided _wrong_ information. It's worse than everything, because they may be building their workflow on a brittle tool, and this always gets back to us as a broken feature and we have to fix it quick. And we do not have the resources for this.
Also, having brittle side-tools can ruin the confidence users would place in `geosx`. I do not think we need this.

> Generation is only disabled with CUDA to avoid producing diffs on LvArray type names (something developers care about), but users need the schema regardless (whether freshly generated or pre-committed) for input validation.

Then if the `schema.xsd` gets generated nevertheless in the installation folder but not into the `git` working tree, this should be fine? End-users would not care about the `LvArray` discrepancy? And this would generate no diff in `git`.

I'm simply saying that the committed `schema.xsd` cannot be considered as a trustworthy source of information.
I'd like `geosx` to produce it for consistency.
As such I'm fine having an installation step which basically does `geosx -s > /path/to/install/schema.xsd`.

> I agree that neither solution is perfect and we should ideally improve the schema generation to not produce diffs between systems/configurations.

I think that the whole generated-doc-in-the-git pattern should be revised.
If the only reason why we do this is `readthedocs`, then we should bundle the generated doc and get `readthedocs` download and deploy it.

Reply:
What I agree with you is that's a bad pattern to commit files that can be generated. 
For me it is to a specific CI step to generate the documentation files and push them into a `documentation` git repository for `readthedocs` to find its files. I have each time a bunch of rst files to commit without really understanding the reason. It gets the commits messy IHMO.

For the time being I will follow your idea: add a specific cmake target that generates always the schema to install it